Plugins/GitUIPluginInterfaces/IGitItem.cs

```diff
@@ -6,6 +6,10 @@ public interface IGitItem
     {
         ObjectId? ObjectId { get; }
         string? Guid { get; }
+    }
+
+    public interface INamedGitItem : IGitItem
```

Comment:
Please move to own file.

Reply:
If this is required, let's add an analyzer for this. There are loads of violations of this today.

Reply:
There's plenty of legacy code that doesn't meet the current standards. But it doesn't mean we should add to the tech debt pile.

As far as analyzers are concerned I'm not sure there's a good one for this case, it is a bit more nuanced. In my experience an interface should be in the same file with a concrete class, unless there is more than one concrete implementation (this is for a typical IoC scenario). Having an interface in the same file significantly reduces maintenance as generally an interface and class change together.
Re# by default offer to do that when extracting an interface, whilst VS by default would place an interface in its own file, and every time I had to use this approach it was a constant context switching nightmare.

On a flipside, if an interface not in the same file, I'd typically look for a file with the same name as the interface. Hence having an interface (or any type for that matter) hidden inside some other file becomes a maintenance overhead because I don't have a luxury of browsing code in an IDE all the time.

Reply:
https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/master/documentation/SA1402.md will enforce this.

I'd vote for enforcing this with that analyzer to save time during reviews. I've moved my branch on from this PR and the cost of switching back to tweak it is an (admittendly minor) barrier to contribution.

Reply:
About 90 occurrences right now, maybe 50 files.
I am not in favor of this rule at all though. It causes code to move around so blame is much harder.
It especially adds an hurdle in adding data classes to group information causing a lot of separate but related status variables or tuples.

But if this is a code style recommendation, I rather use an enforcing rule.

Reply:
> [https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/master/documentation/SA1402.md](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/master/documentation/SA1402.md?rgh-link-date=2021-01-31T11%3A13%3A50Z) will enforce this.
> 
> I'd vote for enforcing this with that analyzer to save time during reviews.

Works for me 👍